tests/core_tests/test_api.py

```diff
@@ -469,20 +469,48 @@ def test_tank_temp(self):
         self.assertRaises(TankError, tank.tank_from_path, self.tank_temp)
 
 
+class TestTankFromEntityWithMixedSlashes(TankTestBase):
+    """
+    Tests the case where a windows local storage uses forward slashes.
+    """
+
+    def test_with_mixed_slashes(self):
+        """
+        Check that a sgtk init works for this path
+        """
+        # only run this test on windows
+        if sys.platform == "win32":
+
+            self.sg_pc_entity["windows_path"] = self.pipeline_config_root.replace("\\", "/")
```

Comment:
Should this test be extended to include testing this?

https://github.com/shotgunsoftware/tk-core/blob/master/tests/core_tests/test_api.py#L472

Maybe we could have a shared base class where you can control both pipeline config paths (with varying slash directions) and storage defs (with missing slashes or mixed slash directions. It would be awesome if the test then probed for a couple of different cases:

- tank_from_path for the pipeline config
- tank_from_path for the project root
- tank_from_path for a shot path
- tank_from_entity for some entity

I think this would cover a whole bunch of init logic in one go. 

Just a thought, if you feel this seems to complicated or not relevant, feel free to ignore!